src/server/pps/server/api_server.go

```diff
@@ -1510,14 +1510,17 @@ func (a *apiServer) getLogsFromStats(pachClient *client.APIClient, request *pps.
 }
 
 func (a *apiServer) validatePipeline(pachClient *client.APIClient, pipelineInfo *pps.PipelineInfo) error {
-	if pipelineInfo.Pipeline == nil {
+	if pipelineInfo.Pipeline == nil || len(pipelineInfo.Pipeline.Name) < 1 {
 		return fmt.Errorf("pipeline has no name")
 	}
 	if err := ancestry.ValidateName(pipelineInfo.Pipeline.Name); err != nil {
-		return fmt.Errorf("Invalid pipeline name: %v", err)
+		return fmt.Errorf("invalid pipeline name: %v", err)
 	}
-	if len(pipelineInfo.Pipeline.Name) > 63 {
-		return fmt.Errorf("Pipeline name is longer than 63 characters: ", len(pipelineInfo.Pipeline.Name))
+	if pipelineInfo.Pipeline.Name[0] == '-' {
+		return fmt.Errorf("pipeline names cannot start with a '-'")
+	}
+	if len(pipelineInfo.Pipeline.Name) > 37 {
```

Comment:
Where does 37 come from?

Reply:
Joey is experiencing this limit (it errors if he tries above), but it is working fine for me, so I'm going to leave this at 63 for now